### DISCUSSION TITLE
Getting "net_context_get(): -2" "Cannot allocate a new TCP connection" log on multiple HTTPS requests

### DISCUSSION THREAD

**Discussion by rajolegajendra-eaton** [zephyrproject-rtos/zephyr#79923] [Q&A]
We are running a webserver. Whenever client sends multiple requests I can see following log on console,

> <err> net_tcp: net_context_get(): -2
> <err> net_tcp: Cannot allocate a new TCP connection

We have this two kconfigs related to TCP connections CONFIG_NET_MAX_CONN and CONFIG_NET_MAX_CONTEXTS. I want to understand how these kconfigs are related to each other? To modify it what should be the ratio? Can someone guide here.

**Comment by jukkar** [comment]
The `CONFIG_NET_MAX_CONTEXTS` corresponds to number of sockets in the system.
The `CONFIG_NET_MAX_CONN` is related to how many active connections there can be at the same time.
They are related to each other and depending on your use cases, they could be kept in sync.
Also important value is `CONFIG_ZVFS_POLL_MAX` that tells how many sockets can be polled (waiting data) at the same time.
For HTTP server, you need to also make sure that `CONFIG_HTTP_SERVER_NUM_SERVICES` and `CONFIG_HTTP_SERVER_MAX_CLIENTS` are set properly (it is used to calculate the number of polled sockets so has relation to `CONFIG_ZVFS_POLL_MAX`)

**Reply by rajolegajendra-eaton** [reply]
Thanks @jukkar . I have one observation TCP connection release is taking time even after client has received complete response. In console I can see it is taking 5-8secs. We are using Zephyr 3.6. How we can reduce this?

**Reply by jukkar** [reply]
See https://docs.zephyrproject.org/latest/connectivity/networking/net_config_guide.html for description of these config options. You can tweak `CONFIG_NET_TCP_TIME_WAIT_DELAY` how long the system waits after the connection is closed.
